barretenberg/cpp/src/barretenberg/stdlib/primitives/padding_indicator_array/padding_indicator_array.hpp

```diff
@@ -34,21 +34,20 @@ namespace bb::stdlib {
  * \f$ [b_0(x-1), \ldots, b_{N-1}(x-1)] \f$ only depends on \f$ N \f$ adding ~\f$ 4\cdot N \f$ gates to the circuit.
  *
  */
-template <typename Fr, typename Builder, size_t virtual_log_n>
-static std::array<Fr, virtual_log_n> compute_padding_indicator_array(const Fr& log_n)
+template <typename Curve, size_t virtual_log_n>
+static std::array<typename Curve::ScalarField, virtual_log_n> compute_padding_indicator_array(
```

Comment:
Again not a blocking point but I think the more idiomatic pattern here would be to make this a class `PaddingIndicatorArray` with a constructor that contains this logic. Maybe the constructor even takes `n` and `constrain_log_circuit_size` (now a class method) gets called automatically under the hood so it can't be missed